README.md
**Cultibox**
============

La Cultibox est un boitier de contrôle du climat permettant de piloter des prises 
avec des programmes définits ou de réagir en fonction des informations remontées
par les capteurs connectés.
Toutes les informations nécéssaires disponibles sur le site officiel:

 [http://www.cultibox.fr](http://www.cultibox.fr)



Pour plus d'informations sur le logiciel open source permettant de piloter le boitier, 
rendez-vous:

 * sur la [documentation utilisateur](https://github.com/cultibox/cultibox/wiki) 
 * sur la [liste des tickets (Amélioration, bug ...)](https://github.com/cultibox/cultibox/issues)
 * sur le [dépôt du code source](https://github.com/cultibox/cultibox/)


Quoi de neuf?
 * Liste des [évolutions pour le logiciel](https://github.com/cultibox/cultibox/blob/master/01_software/CHANGELOG)
 * Lise des [évolutions pour le firmware](https://github.com/cultibox/cultibox/blob/master/01_software/01_install/01_src/03_sd/version.txt)


Et concrètement?
 * Pour récupérer le dépôt principal avec git:

`git clone --recursive https://github.com/cultibox/cultibox.git`
 * Pour récupérer le wiki avec git: 

`git clone --recursive https://github.com/cultibox/cultibox.wiki.git`


01_software/CHANGELOG
Cultibox (2.0.49) precise; urgency=low
    + Bouton de passage en mode Wifi
    + Diverses corrections

-- Yann Genevois <yann.genevois@cultibox.fr> Thu Jun 02 17:25:12 CEST 2015


Cultibox (2.0.48) precise; urgency=low
    + Réorganisation des onglets
    + Fusion programme/prises
    + Ajout des fonctionnalités dans des icones cliquables
    + Ajout de la page cost comme icone
    + Correction de l'aide et ajout d'un css spécifique à l'aide
    + Correction du wizard
    + Correction des webcams si aucune webcam n'est branchée
    + Ajout d'un bouton pour uploader ses images dans le cultipi
    + Programme des prises dans une boite de dialogue
    + Corrections diverses

-- Yann Genevois <yann.genevois@cultibox.fr> Thu Apr 16 12:25:31 CEST 2015


Cultibox (2.0.47) precise; urgency=low
    + Ajout de plusieurs webcams

-- Yann Genevois <yann.genevois@cultibox.fr> Mon Mar 30 18:59:17 CEST 2015


Cultibox (2.0.46) precise; urgency=low
    + Suppréssion des statistiques pour le mode Cultipi

-- Yann Genevois <yann.genevois@cultibox.fr> Sat Mar 14 18:34:14 CEST 2015


Cultibox (2.0.45) precise; urgency=low
    + Remise en forme de l'icône de documentation et des textes de traduction
    + Centrage du menu
    + Gestion erreur synoptique

-- Yann Genevois <yann.genevois@cultibox.fr> Tue Mar 10 14:47:12 CEST 2015


Cultibox (2.0.44) precise; urgency=low
    + Correction du minimum des axes Y sur les logs
    + Correction d'une requêtte sur les prises

-- Yann Genevois <yann.genevois@cultibox.fr> Sun Mar 08 22:07:43 CEST 2015


Cultibox (2.0.43) precise; urgency=low
    + Correction export pour Mac OS X

-- Yann Genevois <yann.genevois@cultibox.fr> Thu Feb 24 19:50:16 CEST 2015


Cultibox (2.0.42) precise; urgency=low
    + Reset et création de la configuration réseau sans redémarrage

-- Yann Genevois <yann.genevois@cultibox.fr> Thu Feb 24 19:50:16 CEST 2015


Cultibox (2.0.41) precise; urgency=low
    + Ajout d'une passerelle pour es adressages statiques

-- Yann Genevois <yann.genevois@cultibox.fr> Sun Feb 22 16:33:12 CEST 2015


Cultibox (2.0.40) precise; urgency=low
    + Correction de l'affichage des logs

-- Yann Genevois <yann.genevois@cultibox.fr> Sat Feb 21 14:44:23 CEST 2015


Cultibox (2.0.39) precise; urgency=low
    + Correction de l'affichage des logs
    + Ajout d'un dépôt dev
    + Correction de l'export des logs
    + Correction de la gestion des mises à jours
    + Gestion des fichiers gzip pour l'import de configuration

-- Yann Genevois <yann.genevois@cultibox.fr> Sat Feb 21 11:38:31 CEST 2015


Cultibox (2.0.38) precise; urgency=low
    + Correction de la gestion du réseau

-- Yann Genevois <yann.genevois@cultibox.fr> Sun Feb 15 14:40:35 CEST 2015


Cultibox (2.0.37) precise; urgency=low
    + Ajout de la configuration de la résolution de la webcam
    + Correction de l'affichage de synoptique et de l'import de configuration
    + Utilisation du format zip pour l'export/import de configuration Cultipi
    + Configuration de la date en mode adHoc
    + Navigation possible en adHoc

-- Yann Genevois <yann.genevois@cultibox.fr> Sat Feb 14 11:35:12 CEST 2015


Cultibox (2.0.36) precise; urgency=low
    + Correction du focus pour la console
    + Ajout des clefs hexadécimal pour WEP
    + Ajout des clefs 64bits WEP
    + Correction du fichier de configuration WEP
    + Ajout de la gestion de brightness et contrast

-- Yann Genevois <yann.genevois@cultibox.fr> Tue Feb 12 10:15:28 CEST 2015


Cultibox (2.0.35) precise; urgency=low
    + Correction du décallage de l'heure de prise de l'image webcam
    + Correction de la valeur de la seconde tolérance 
    + Masque par défaut à 255.255.255.0 si non définit
    + Correction de l'affichage de la partie wifi
    + Suppréssion des options inutiles pour le wifi
    + Suppréssion des scripts qui tournent en boucle avec le setTimeout ou setInterval
    + Ajout d'un bouton de RAZ de la configuraiton réseau
    + Ajout d'un bouton de téléchargement du service Bonjour

-- Yann Genevois <yann.genevois@cultibox.fr> Thu Feb 10 16:16:35 CEST 2015


Cultibox (2.0.34) precise; urgency=low
    + Corrections création image webcam
    + Correction bouton enregistrer la configuration
    + Correction locales

-- Yann Genevois <yann.genevois@cultibox.fr> Thu Feb 05 11:42:46 CEST 2015


Cultibox (2.0.33) precise; urgency=low
    + Corrections import/export pour Windows

-- Yann Genevois <yann.genevois@cultibox.fr> Wed Feb 04 11:46:28 CEST 2015


Cultibox (2.0.32) precise; urgency=low
    + Import/export d'une configuration

-- Yann Genevois <yann.genevois@cultibox.fr> Tue Feb 03 15:43:12 CEST 2015


Cultibox (2.0.31) precise; urgency=low
    + Fenêtre d'attente de chargement des pages

-- Yann Genevois <yann.genevois@cultibox.fr> Mon Feb 02 18:24:56 CEST 2015


Cultibox (2.0.30) precise; urgency=low
    + Possibilité de voir et télécharger les logs des services
    + Ajout de l'export de la configuration

-- Yann Genevois <yann.genevois@cultibox.fr> Sat Jan 31 10:42:23 CEST 2015


Cultibox (2.0.29) precise; urgency=low
    + Correction de la prise d'image pour les webcams
    + Suppréssion de la configuration de la couleur des puissances
    + Ajout d'un message à la fin de la synchronisation
    + Correction de la boite de dialgue
    + Correction de la synchronisation
    + Correction du décallage d'une here sur les programmes
    + Fonction d'upgrade des packages

-- Yann Genevois <yann.genevois@cultibox.fr> Thu Jan 29 12:12:34 CEST 2015


Cultibox (2.0.28) precise; urgency=low
    + Remise en forme de l'administration du Cultipi
    + Ajout des infos de régulation secondaire au chargement de la page programme
    + Ajout du temps écoulé et restant d'un évènement par rapport à la date courante dans le calendrier
    + Ajout d'un bouton de redémarrage du module

-- Yann Genevois <yann.genevois@cultibox.fr> Wed Jan 21 18:31:14 CEST 2015


Cultibox (2.0.27) precise; urgency=low
    + Correction de l'affichage du lien synoptique dans la page welcome

02_documentation/06_engrais/engrais.tcl
set outPutDir [file join [file dirname [info script] ] out]
set outPutDoc [file join [file dirname [info script] ] gui_soft_cal_engrais.wiki]

set inputFile [file join [file dirname [info script] ] engrais.txt]
set inputFileRef [file join [file dirname [info script] ] engrais_ref.txt]


proc initFile {fid marque substrat programme} {
	puts $fid {<?xml version="1.0" encoding="utf-8"?>}
	puts $fid {<feed xmlns="http://www.w3.org/2005/Atom">}
	puts $fid "   <updated>[clock format [clock seconds] -format {%Y-%m-%dT%H:%M:%S+02:00}]</updated>"
	puts $fid {   <id>http://www.cultibox.fr/</id>}
	puts $fid "   <title>${marque} ${programme} ${substrat}</title>"
	puts $fid {   <subtitle></subtitle>}
	puts $fid {   <author>}
	puts $fid {      <name>Cultibox</name>}
	puts $fid {      <uri>http://www.cultibox.fr/</uri>}
	puts $fid {      <email>info@cultibox.fr</email>}
	puts $fid {   </author>}
	puts $fid {   <category term = "engrais" label = "engrais">}
	puts $fid "    <substrat>${substrat}</substrat>"
	puts $fid "    <periode>${programme}</periode>"
	puts $fid "    <marque>${marque}</marque>"
	puts $fid {   </category>}
	puts $fid "   <link rel=\"self\" href=\"www.cultibox.fr/${marque}_${substrat}.xml\" />"
	puts $fid {   <icon></icon>}
	puts $fid {   <logo></logo>}
	puts $fid {   <rights type = "text">}
	puts $fid {    © Cultibox, 2013}
	puts $fid {   </rights>}
}

proc closFile {fid} {
	puts $fid {</feed>}
}

proc addNutriment {fid name dose} {
	puts $fid {      <nutriment>}
	puts $fid "        <name>${name}</name>"
	puts $fid "        <dosage>${dose}</dosage>"
	puts $fid {      </nutriment>}
}

proc startEntry {fid start name remarque ec facultative} {
	puts $fid {    <entry>}
	puts $fid "      <title>S[expr ${start} / 7 + 1] ${name}</title>"
	puts $fid "      <summary>S[expr ${start} / 7 + 1] ${name}</summary>"
	puts $fid {      <updated>2008-05-29T13:14:57+02:00</updated>}
	puts $fid {      <id>http://www.cultibox.fr/id1</id>}
	puts $fid {      <category term = "vegetation" label = "Vegetation" />}
	puts $fid "      <content type = \"text\">${remarque}</content>"
	puts $fid "      <ec>${ec}</ec>"
	puts $fid {      <duration>7</duration>}
	puts $fid "      <start>${start}</start>"
	puts $fid "      <facultative>${facultative}</facultative>"
}

proc closeEntry {fid marque color} {
	puts $fid "      <color>${color}</color>"
	puts $fid "      <icon>[string map {" " "_"} ${marque}].png</icon>"
	puts $fid {   </entry>}
}

set fidIn [open $inputFile r]

set oldProgramme ""
set oldSubstrat ""

set fid ""
gets $fidIn UneLigne
set EngraisMarque [split $UneLigne "\t"]

gets $fidIn UneLigne
set EngraisName [split $UneLigne "\t"]


while {[eof $fidIn] != 1} {
	gets $fidIn UneLigne
	set UneLigne [split $UneLigne "\t"]
	
	set marque [lindex $UneLigne 0]
	set nomProgramme [lindex $UneLigne 1]
	set Substrat  [lindex $UneLigne 2]
	set phase [lindex $UneLigne 3]
	set remarque [lindex $UneLigne 4]
	set ec [lindex $UneLigne 5]
    set utilise [lindex $UneLigne 6]
	set facultative [lindex $UneLigne 7]
	
    if {$utilise == 1} {

        if {$nomProgramme != $oldProgramme || $Substrat != $oldSubstrat} {
            if {$fid != ""} {
                closFile $fid
                close $fid
                set fid ""
            }
            if {$nomProgramme != ""} {
                set fid [open [file join $outPutDir "[string map {" " "_"} ${marque}]_[string map {" " "_"} ${nomProgramme}]_[string map {" " "_"} $Substrat].xml"] w+]
                initFile $fid $marque ${Substrat} $nomProgramme
            }
        }
        
        if {$fid != ""} {

            
        
            switch $phase {
                "S1" -
                "Graines" {
                    set start 0
                    set phase Croissance
                    set color "#98FB98"
                }
                "S2" -
                "Plantules" {
                    set start 7
                    set phase Croissance
                    set color "#9ACD32"
                }
                "S3" -
                "Boutures" {
                    set start 14
                    set phase Croissance
                    set color "#006400"
                }
                "S4" -
                "Veg 18 h" {
                    set start 21
                    set phase Floraison
                    set color "#F0E68C"
                }
                "S5" -
                "Pass 12/12 " {
                    set start 28
                    set phase Floraison
                    set color "#F5DEB3"
                }
                "S6" -
                "Strech" {
                    set start 35
                    set phase Floraison
                    set color "#D2B48C"
                }		
                "S7" -
                "1ers pistils" {
                    set start 42
                    set phase Floraison
                    set color "#DAA520"
                }		
                "S8" -			
                "Debut Flo" {
                    set start 49
                    set phase Floraison
                    set color #CD853F
                }
                "S9" -
                "Floraison" {
                    set start 56
                    set phase Floraison
                    set color #D2691E
                }
                "S10" -
                "Boom floral" {
                    set start 63
                    set phase Floraison
                    set color #A0522D
                }
                "S11" -
                "Fin floraison" {
                    set start 70
                    set phase Floraison
                    set color #8B4513
                }
                "S12" -			
                "Noir" {
                    set start 77
                    set phase Rincage
					set remarque Rincage
                    set color #000000
                }			
            }

            startEntry $fid $start $phase $remarque $ec $facultative
            
            set idx 8
            foreach Dosage [lrange $UneLigne 8 end] {
                
                if {$Dosage != 0} {
                    set unit "ml/l"
					switch [lindex $EngraisName $idx] {
						"Mineral Magic" -
						"Piranha" -
						"Carboload" -
						"Bud Blood" -
						"Big Bud" {
							set unit "g/l"
						}
						"SuperVit" {
							set unit "goutte/4,5l"
						}
					}
                    addNutriment $fid [lindex $EngraisName $idx] "${Dosage} ${unit}"
                }
                incr idx
            }
            closeEntry $fid $marque $color
        }
        
        
        set oldProgramme $nomProgramme
        set oldSubstrat $Substrat
    
    }
	
	
}

if {$fid != ""} {
	closFile $fid
	close $fid
	set fid ""
}

close $fidIn


# Génération de la doc

# Chargement du fichier de référence sur les engrais
set fidIn [open $inputFileRef r]
gets $fidIn UneLigne
while {[eof $fidIn] != 1} {
	gets $fidIn UneLigne
    set UneLigne [split $UneLigne "\t"]
    set marque [lindex $UneLigne 0]
    set utilise [lindex $UneLigne 1]
	set source [lindex $UneLigne 2]
	set engref(${marque},source) $source
}
close $fidIn


set fid [open $outPutDoc w+]
set fidIn [open $inputFile r]

gets $fidIn UneLigne
set EngraisMarque [split $UneLigne "\t"]

gets $fidIn UneLigne
set EngraisName [split $UneLigne "\t"]

puts $fid {#summary Calendrier des engrais

= Sommaire =
<wiki:toc max_depth="3" />

= Introduction =

Le logiciel Cultibox vous permet de planifier les engrais que vous souhaitez appliquer.

La douxième semaine est systématiquement une semaine de rinçage.

Ci-dessous vous trouverez tout les schémas de culture utilisés :

<wiki:comment>
Disponible en annexe de la documentation.
-annexe-
\chapter{Tableau des engrais}  \label{nutrients_tab}
\begin{landscape}
</wiki:comment>

}

set oldProgramme ""
set oldSubstrat ""
set oldmarque ""
set semaine 1
    
while {[eof $fidIn] != 1} {
	gets $fidIn UneLigne
    set UneLigne [split $UneLigne "\t"]
    set marque [lindex $UneLigne 0]
    set nomProgramme [lindex $UneLigne 1]
    set Substrat  [lindex $UneLigne 2]
    set phase [lindex $UneLigne 3]
    set remarque [lindex $UneLigne 4]
    set ec [lindex $UneLigne 5]
    set utilise [lindex $UneLigne 6]
        
    
        
    if {$utilise == 1 || [eof $fidIn]} {
    
        # Si ça change on écrit le vecteur
        if {$Substrat != $oldSubstrat || $marque != $oldmarque || $nomProgramme != $oldProgramme } {
            if {[array exists eng]} {
                set change 0
                
                set listEngrais ""
                set listSemaine ""
                foreach name [array names eng] {
                    if {[lindex [split $name ","] 1] != "remarque" && [lindex [split $name ","] 1] != "ec"} { 
                    lappend listSemaine [lindex [split $name ","] 0]
                    lappend listEngrais [lindex [split $name ","] 1]
                    }
                }
                set listEngrais [lsort -unique $listEngrais]
                set listSemaine [lsort -integer -unique $listSemaine]
                
                #-- Header du tableau --
                puts -nonewline $fid "|| *Semaine* "
                
                # On cherche si une note est présente
                set NOTEpresent 0
                foreach Semaine $listSemaine {
                    if {$eng(${Semaine},remarque) != ""} {
                        incr NOTEpresent
                        break;
                    }
                }
                if {$NOTEpresent != 0} {
                    puts -nonewline $fid "|| *Note* "
                }
                
                # On cherche si un EC est présent
                set ECpresent 0
                foreach Semaine $listSemaine {
                    if {$eng(${Semaine},ec) != ""} {
                        incr ECpresent
                        break;
                    }
                }
                if {$ECpresent != 0} {
                    puts -nonewline $fid "|| *EC* "
                }

                # On ajoute dans le header la liste des engrais
                foreach Engrais $listEngrais {
                    puts -nonewline $fid "|| *${Engrais}* "
                }
                puts -nonewline $fid "||"
                puts $fid ""
                    
                foreach Semaine $listSemaine {
                    puts -nonewline $fid "|| $Semaine "
                    
                    # Si au moins une note est présent
                    if {$NOTEpresent != 0} {
                        puts -nonewline $fid "|| $eng(${Semaine},remarque) "
                    } 
                    
                    # Si au moins un EC est présent
                    if {$ECpresent != 0} {
                        puts -nonewline $fid "|| $eng(${Semaine},ec) "
                    } 
                    
                    foreach Engrais $listEngrais {
                        if {[array names eng ${Semaine},${Engrais}] != ""} {
                            puts -nonewline $fid "|| $eng(${Semaine},${Engrais})"
                        } else {
                            puts -nonewline $fid "||  "
                        }
                    }
                    puts -nonewline $fid "||"
                    puts $fid ""
                }
				puts $fid ""
                array unset eng
                

            }
        }
    
        if {$marque != $oldmarque && $marque != ""} {
            puts $fid "= ${marque} ="
            puts $fid ""
			
			if {[array names engref ${marque},source] != ""} {
				puts $fid "Source : \[$engref(${marque},source) Schéma de culture\]"
				puts $fid ""
			}
			
        }
        
        if {($nomProgramme != $oldProgramme && $nomProgramme != "") || ($Substrat != $oldSubstrat && $Substrat != "")} {
            puts $fid "== ${nomProgramme} - ${Substrat} =="
            puts $fid ""
			set oldSubstrat ""
            set semaine 1
        }


        set idx 7
        set eng(${semaine},ec) $ec
        if {$remarque != "Rincage"} {
            set eng(${semaine},remarque) $remarque
        } else {
            set eng(${semaine},remarque) ""
        }
        foreach Dosage [lrange $UneLigne 7 end] {
            
            if {$Dosage != 0} {
                set unit "ml/l"
                if {[lindex $EngraisName $idx] == "Mineral Magic" || [lindex $EngraisName $idx] == "Piranha"} {
                    set unit "g/l"
                }
				if {[lindex $EngraisName $idx] == "SuperVit"} {
					set unit "goutte / 4,5l"
				}
                set eng(${semaine},[lindex $EngraisName $idx]) "${Dosage} ${unit}"
            }
            incr idx
        }
        incr semaine
        
                set oldProgramme $nomProgramme
                set oldSubstrat $Substrat
                set oldmarque $marque
    }


}

puts $fid {

<wiki:comment>
\end{landscape}
-annexeend-
</wiki:comment>

}

close $fid
close $fidIn


